featuretools/synthesis/deep_feature_synthesis.py

```diff
@@ -208,25 +208,20 @@ def build_features(self, variable_types=None, verbose=False):
                 all_features[e.id] = {}
 
         self.where_clauses = defaultdict(set)
+
+        if return_variable_types is None:
```

Comment:
let's move this logic inside of if block on line 235. that way it is closer to where it's used and we don't have the double check for `return_variable_types == 'all'` case

Reply:
Isn't the assertion meant to provide a check for back input and give feedback? To do that, it needs to go before `self._run_dfs`.

Reply:
ahh, i didn't see the assert. okay. let's leave as is